apps/support/Makefile.inc

```diff
@@ -130,13 +143,13 @@ endif
 $(BIN)/%.registration.cpp: $(BIN)/%.a
 	@echo $@ produced implicitly by $^
 
-$(BIN)/RunGenMain.o: $(HALIDE_DISTRIB_PATH)/tools/RunGenMain.cpp $(HALIDE_DISTRIB_PATH)/tools/RunGen.h
+$(BIN)/%/RunGenMain.o: $(HALIDE_DISTRIB_PATH)/tools/RunGenMain.cpp $(HALIDE_DISTRIB_PATH)/tools/RunGen.h
 	@mkdir -p $(@D)
-	@$(CXX) -c $< $(CXXFLAGS) $(IMAGE_IO_CXX_FLAGS) -I$(BIN) -o $@
+	@$(CXX) -c $< $(CXXFLAGS) $(IMAGE_IO_CXX_FLAGS) -I$(BIN)/$* -o $@
 
 # Really, .SECONDARY is what we want, but it won't accept wildcards
-.PRECIOUS: $(BIN)/%.rungen
-$(BIN)/%.rungen: $(BIN)/RunGenMain.o $(BIN)/%.a $(BIN)/%.registration.cpp
+.PRECIOUS: $(BIN)/$(HL_TARGET)/%.rungen
+$(BIN)/$(HL_TARGET)/%.rungen: $(BIN)/$(HL_TARGET)/RunGenMain.o $(BIN)/$(HL_TARGET)/%.a $(BIN)/$(HL_TARGET)/%.registration.cpp
```

Comment:
I don't think RunGenMain.o actually depends on the halide target, so how about: 
```
$(BIN)/%.rungen: $(BIN)/RunGenMain.o $(BIN)/%.a $(BIN)/%.registration.cpp
```

Reply:
Actually, it will, for the Wasm case that I'm working on :-)

Reply:
Nuts, I was hoping to avoid multiple wildcards to make the rungen rule below work. I guess another alternative is to just have the name of rungen depend on the filter name: 
```
$(BIN)/%_RunGenMain.o: rungen sources
   ... use string manipulation to extract the target from $*

$(BIN)/%.rungen: $(BIN)/%_RunGenMain.o $(BIN)/%.a $(BIN)/%.registration.cpp
```

Reply:
Alternatively:

```
$(BIN)/%_RunGenMain.o: $(BIN)/RunGenMain.o
   ... to avoid repeatedly recompiling RunGenMain, either just cp $(BIN)/RunGenMain.o or do something fancier for wasm

$(BIN)/%.rungen: $(BIN)/%_RunGenMain.o $(BIN)/%.a $(BIN)/%.registration.cpp
```

Reply:
let's hold off on that, since the actual implementation of the wasm stuff might change this

Reply:
If we're going to hold off, why not use the simpler first suggestion with a single RunGenMain.o? If I'm understanding things correctly, in the current state the following will produce a confusing error about a make target not existing:

```
make bin/x86-64-linux/local_laplacian.run
```

this is what we have in master, but with this PR it will produce a similar error:
```
HL_TARGET=x86-64-linux make bin/local_laplacian.run
```

You'd need to say:
```
HL_TARGET=x86-64-linux make bin/x86-64-linux/local_laplacian.run
```

make targets not existing is very annoying to debug

Reply:
OK, just pushed a revision that I think will work; it basically relies on % being greedy. PTAL